### PR TITLE
[RPRBLND-1667] Support Principled Hair BSDF shader node

### PR DESCRIPTION
**Pull request by ardraeiss**
### PURPOSE
Added basic support for Cycles Principled Hair BSDF shader node.

### EFFECT OF CHANGE
- added basic support for Cycles Principled Hair BSDF shader node. Randomness is not supported. The best results are in the "Direct" coloring mode.

### TECHNICAL STEPS
- added parameters-to-absorption conversions using Cycles behavior and conversion formulas;
- added shader approximation using Uber node and absorption and roughness arguments;
- added Principled Hair BSDF node to the shader nodes menu.

### NOTES FOR REVIEWERS
- the Absorption and Melanine results are brighter that what Cycles does;
- Tint in Melanine Concentration mode isn't supported yet;
- reflection Roughness is very roughly approximated;

### CODE REVIEW

File: src/rprblender/nodes/blender_nodes.py [hunk 1]
```diff
@@ -969,6 +969,77 @@ def _create_transmission_node(self, base_color, roughness):
...
+            color = self.get_input_scalar('Color')
+            absorption_color = self._calculate_absorption_from_color(color, roughness_radial)
+
+        node = self._create_absorption_node(color, absorption_color, roughness, roughness_radial, coat, ior)
+        return node
+
+    def _calculate_absorption_from_color(self, color, roughness_radial):
+        """ Use the Cycles way from bsdf_principled_hair_albedo_roughness_scale and bsdf_principled_hair_sigma_from_reflectance """
+        x = roughness_radial
+        roughness_scale = (((((0.245 * x) + 5.574) * x - 10.73) * x + 2.532) * x - 0.215) * x + 5.969
```
Comment (bsavery) [inline src/rprblender/nodes/blender_nodes.py:1001]: Where did this calculation come from

File: src/rprblender/nodes/blender_nodes.py [hunk 2]
```diff
@@ -969,6 +969,77 @@ def _create_transmission_node(self, base_color, roughness):
...
+        color_log = self.node_item(tuple(math.log(e, 10) for e in color.data))
+        result = color_log / roughness_scale
+        result = result.max(0.0)
+        return result
+
+    def _calculate_absorption_from_melanin(self, melanin, melanin_redness):
+        """ Use the Cycles way from bsdf_principled_hair_sigma_from_concentration """
+        eumelanin = melanin * (1.0 - melanin_redness)
+        pheomelanin = melanin * melanin_redness
+        result = eumelanin * (0.506, 0.841, 1.653) + pheomelanin * (0.343, 0.733, 1.924)
```
Comment (bsavery) [inline src/rprblender/nodes/blender_nodes.py:1011]: also these calculations

File: src/rprblender/nodes/blender_nodes.py (hunk 1 above)
Comment (ardraeiss) [reply src/rprblender/nodes/blender_nodes.py:1001]: Cycles code and Blender Manual https://docs.blender.org/manual/en/latest/render/shader_nodes/shader/hair_principled.html

File: src/rprblender/nodes/blender_nodes.py (hunk 2 above)
Comment (ardraeiss) [reply src/rprblender/nodes/blender_nodes.py:1011]: Same source, the Blender Manual https://docs.blender.org/manual/en/latest/render/shader_nodes/shader/hair_principled.html

**Review by bnagirniak — APPROVED** [review]
Tested - works ok.

File: src/rprblender/nodes/blender_nodes.py [hunk 3]
```diff
@@ -969,6 +969,79 @@ def _create_transmission_node(self, base_color, roughness):
...
+
+    def _create_absorption_node(self, color, absorption_color, roughness, roughness_radial, coat, ior):
+        base_color = (1.0 - absorption_color)
+
+        rpr_node = self.create_node(pyrpr.MATERIAL_NODE_UBERV2)
+        rpr_node.set_input(pyrpr.MATERIAL_INPUT_UBER_DIFFUSE_WEIGHT, 0.0)
+
+        rpr_node.set_input(pyrpr.MATERIAL_INPUT_UBER_BACKSCATTER_WEIGHT, 1.0)
+        rpr_node.set_input(pyrpr.MATERIAL_INPUT_UBER_BACKSCATTER_COLOR, color)
+        rpr_node.set_input(pyrpr.MATERIAL_INPUT_UBER_SSS_WEIGHT, 1.0)
```
Comment (bsavery) [inline src/rprblender/nodes/blender_nodes.py:1024]: I think just use backscatter not SSS.  For lighter weight computation.  Does that look ok visually?

**Review by bsavery — CHANGES_REQUESTED** [review]
Just use Backscatter if visually is ok.

File: src/rprblender/nodes/blender_nodes.py (hunk 3 above)
Comment (ardraeiss) [reply src/rprblender/nodes/blender_nodes.py]: Yes, the "with SSS" and "without SSS" do look similar, at least on hairs.

**Pull request merged**